kmath-core/src/commonMain/kotlin/space/kscience/kmath/operations/LogicAlgebra.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.operations
+
+import space.kscience.kmath.expressions.Symbol
+import space.kscience.kmath.expressions.symbol
+import space.kscience.kmath.misc.UnstableKMathAPI
+
+/**
+ * An algebra for generic boolean logic
+ */
+@UnstableKMathAPI
+public interface LogicAlgebra<T : Any> : Algebra<T> {
```

Comment:
Are there any ways to implement this interface otherwise than with `Boolean`?

Reply:
Yes, obviously we have MstLogicAlgebra and there could be boolean vectors, bit-masks, etc

Reply:
OK then